### PR TITLE
Cleanup device interaction

### PR DESCRIPTION
**Pull request by SuperSolik**

### CODE REVIEW

File: device_interaction.py [hunk 1]
```diff
@@ -33,13 +33,14 @@ def read_page_data(pid):
...
+            # Convert to uint64_t
             offset = struct.unpack('Q', offset)[0]
+            # Convert to uint32_t
             flags = struct.unpack('I', flags)[0]
-            data.append([offset,  # Swap offset or page address
+            data.append((offset,  # Swap offset or page address
                          get_bit(flags, shift=26),  # return 1 if page present : 0 if swapped
                          get_bit(flags, shift=4),  # return 1 if page dirty : 0 if clean
-                         get_bit(flags, shift=12)])  # return 1 if page anonymous : 0 if not
+                         get_bit(flags, shift=12)))  # return 1 if page anonymous : 0 if not
```
Comment (OlgaKochneva) [inline device_interaction.py:43]: Is there any difference between ] и )?

File: device_interaction.py [hunk 2]
```diff
@@ -117,16 +119,16 @@ def __init__(self):
         self.present_page_data = {}
         self.swapped_page_data = {}
         self.error_pids = []
-        self.all_pid_list = []
-        self.cgroup_pid_list = []
+        self.pid_list_all = []
+        self.pid_list_cgroup = []
```
Comment (OlgaKochneva) [inline device_interaction.py:123]: I guess it is not so important, it is more about preferences. You can leave it

File: device_interaction.py [hunk 3]
```diff
@@ -146,44 +148,45 @@ def set_device(self, device):
...
+            self.pid_list_all = adb_collect_pid_list(self.device,
                                                      tool='get_pid_list',
                                                      filename='pid_list',
                                                      pull_path='pids_data',
-                                                     group='')
+                                                     group_name='')
         except Exception:
-            self.all_pid_list = []
+            self.pid_list_all = []
             raise
```
Comment (OlgaKochneva) [inline device_interaction.py]: `except Exception` looks really bad. But we can't fix it, can we?

File: handling/device_handler.py [hunk 4]
```diff
@@ -1,27 +1,28 @@
...
+
+from utilities import exec_command
 
 
 class DeviceHandler:
     def __init__(self):
         self.listeners = []
         self.serial_numbers = []
-        self.current_device = None
+        self.current_device_number = None
```
Comment (OlgaKochneva) [inline handling/device_handler.py]: I suggest `current_device_id`. It is shorter and closer to serial number

File: device_interaction.py (hunk 1 above)
Comment (SuperSolik) [reply device_interaction.py:43]: () , or tuple, costs less memory than list and more efficient, so when we have an immutable data like in this case, it's better to use tuples

File: handling/device_handler.py (hunk 4 above)
Comment (SuperSolik) [reply handling/device_handler.py]: Good point, thanks

**Review by OlgaKochneva — APPROVED** [review]

**Pull request merged**